packages/web-components/src/badge/badge.template.ts

```diff
@@ -0,0 +1,18 @@
+import { ElementViewTemplate, html } from '@microsoft/fast-element';
+import { endSlotTemplate, startSlotTemplate, staticallyCompose } from '@microsoft/fast-foundation';
+import type { Badge } from './badge.js';
+import type { BadgeOptions } from './badge.options.js';
+
+/**
+ * The template for the Badge component.
+ * @public
+ */
+export function badgeTemplate<T extends Badge>(options: BadgeOptions = {}): ElementViewTemplate<T> {
```

Comment:
This is the first component which export its template function. How do we decide whether we export it or not?
Only when we have a internal use case for it (like in the `counter-badge` here) or always?

Reply:
I think there's value if someone needs to create their own badge or _extend_ the Fluent badge to get all the goodness and not have to recreate. One of the benefits of that would be more shared code and less repeated. Now, in the case that they need to "fork" the template that would result in duplication. I don't think we need to advertise it necessarily, but in terms of enabling teams to reuse and compose, etc...we've found it to be fairly helpful. Text is technically exporting the used template from the index as well.

I'm open.

With that said, in this specific case, it's definitely exported here to enable shared reuse and code for badge + counter badge. Whether we want to export from the index or not we can discuss.

Reply:
I agree with this being exported, if we are not sure, we can keep it private and wait for the demand.
Personally I would prefer every component to export its template :+1: